opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -240,15 +432,23 @@ class WellModelMatrixAdapter : public Dune::AssembledLinearOperator<M,X,Y>
         {
             parameters_.template init<TypeTag>();
             extractParallelGridInformationToISTL(simulator_.vanguard().grid(), parallelInformation_);
+
             auto gridForConn = simulator_.vanguard().grid();
             const auto wellsForConn = simulator_.vanguard().schedule().getWellsatEnd();
             const bool useWellConn = EWOMS_GET_PARAM(TypeTag, bool, MatrixAddWellContributions);
 
-            detail::setWellConnections(gridForConn, wellsForConn, useWellConn, wellConnectionsGraph_);
-            detail::findOverlapAndInterior(gridForConn, overlapRows_, interiorRows_);
-            if (gridForConn.comm().size() > 1) {
-                noGhostAdjacency();
-                setGhostsInNoGhost(*noGhostMat_);
+            ownersFirst_ = EWOMS_GET_PARAM(TypeTag, bool, OwnersFirst);
+            interiorCellNum_ = detail::numInteriorCells(simulator_.vanguard().grid(), ownersFirst_);
+
+            if (!ownersFirst_) {
```

Comment:
setWellConnections, findOverlapAndInterior, noGhostAdjacency and setGhostsInNoGhost are only needed for making copying of the Jacobean possible. With --owners-first=true a copy of the Jacobean is not needed, and information about the sparsity pattern is not needed.

flow --owners-first=true --matrix-add-well-connections=true works fine since all well contributions are located in the interior/owner rows.